### PR TITLE
Azure CosmosDB Vector Search integration

### PR DESCRIPTION
**Pull request by aayush3011**
### Related Issues

Azure Cosmos DB Mongo vCore
- [Vector Search for Mongo vCore](https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search)

Azure Cosmos DB NoSql
- [Vector Search for NoSql](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search)

### Proposed Changes:

Added vector search support for Azure Cosmos DB for NoSql and Mongo vCore. It includes all the features mentioned in the links above.

 
 

### How did you test it?

Added unit tests, and integration tests for testing.

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/azure_cosmos_db_mongo_vcore/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# mongodb-atlas-haystack
+
+[![PyPI - Version](https://img.shields.io/pypi/v/mongodb-atlas-haystack.svg)](https://pypi.org/project/mongodb-atlas-haystack)
```
Comment (julian-risch) [inline integrations/azure_cosmos_db_mongo_vcore/README.md:3]: link to the pypi package here should be the same as in the pip install command

File: integrations/azure_cosmos_db_mongo_vcore/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+**Table of Contents**
+
+- [Installation](#installation)
+- [Contributing](#contributing)
+- [License](#license)
+
+## Installation
+
+```console
+pip install azure_cosmos_db_mongo_vcore-haystack
```
Comment (julian-risch) [inline integrations/azure_cosmos_db_mongo_vcore/README.md:17]: that will more likely be `pip install azure-cosmos-db-mongo-vcore-haystack`

File: integrations/azure_cosmos_db_nosql/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# mongodb-atlas-haystack
+
+[![PyPI - Version](https://img.shields.io/pypi/v/mongodb-atlas-haystack.svg)](https://pypi.org/project/mongodb-atlas-haystack)
```
Comment (julian-risch) [inline integrations/azure_cosmos_db_nosql/README.md:3]: links need to be updated

**Review by julian-risch — COMMENTED** [review]
Hello @aayush3011 and thank you for opening this pull request. Please understand that we'll need some time to review it, it's quite a number of lines of code. By the way, have you considered releasing it yourself in a separate repository? We would still list it in the overview of Haystack integrations https://haystack.deepset.ai/integrations/ but the difference would be that you keep ownership of the code.

**Comment by anakin87** [comment]
It seems that we can already use Azure CosmosDB via our existing integrations.
See https://haystack.deepset.ai/integrations/azure-cosmos-db.

I am closing the PR for the moment.

**Pull request closed without merging**